local.go

```diff
@@ -35,7 +36,20 @@ func NewLocalFilesystemBackend(rootDirectory string) *LocalFilesystemBackend {
 	if err != nil {
 		panic(err)
 	}
-	b := &LocalFilesystemBackend{RootDirectory: absPath}
+	// Create a temporary folder for partially-completed writes (if not present)
+	tempPath := filepath.Join(absPath, ".tmp")
```

Comment:
this temp directory creation might be better suited in the `PutObject` function (like we do with the `RootDirectory`). This way the temp directory will always be recreated if it doesn't exist during a put, that way if that directory was deleted while chartmuseum is running, uploads will continue to work without restarting chartmuseum.

Reply:
That would require an extra `stat` per invocation of PutObject. Do we really want that performance overhead?

Reply:
that is definitely a consideration, but it seems like we should be treating the root dir and this new dir similarly. i.e. either both are created in `NewLocalFilesystemBackend` or both are created in `PutObject`. @jdolitsky @scbizu do you have any thoughts here?

Reply:
For @cbuto 's consideration , how about do anther `os.MkdirAll()` and ignore the dir already exist error in `PutObject` , I agree with cbuto's opinion and it is maybe reasonable even we do some overhead tradeoff here.